tests/test_proxy_functional.py

```diff
@@ -881,6 +882,21 @@ async def xtest_proxy_from_env_https_with_auth(
     assert r2.headers["Proxy-Authorization"] == auth.encode()
 
 
+@pytest.mark.xfail
```

Comment:
What's with the xfail? Surely the test should be working?

Reply:
I don't remember why I put that annotation, I will remove it or try to understand why I add it.

Reply:
After trying to make that test work, I'm getting an SSL error trying to connect to the HTTPs website and get the response. Haven't been able to find any example on how should I be able to test this.

I was able to make it work with the fixture that uses proxy.py, but in that case I'm not able to modify the headers that I receive on the CONNECT reply.

I tried also to make it work with the mock, but I think that that implementation is not able to process HTTPS requests.

Any idea?

Reply:
I think we really want a test that doesn't rely on an external server. @webknjaz @bdraco Any ideas how to rewrite the test?

Reply:
Should we just follow the suite of `test_secure_https_proxy_absolute_path`?

Reply:
It is possible to stick a unique header into `proxy.py` via https://github.com/abhinavsingh/proxy.py#modifyrequestheaderplugin. Here's an example of setting a “sentinel” header that my colleague came up with: https://github.com/ansible/ansible/pull/83085/files#diff-5b303b7467efd4c7529d1e171e39e472261e198fb172a45de566c9ec614824d2R3-R32.